server_src/runners/state.js

```diff
@@ -3272,13 +3274,18 @@ class Runner {
         printer.markModified("apikey");
         updatePrinter = true;
       }
+
+      // Make sure OctoPrint status is updated before continuing on with settings...
+      if (updatePrinter) {
+        await Runner.reScanOcto(farmPrinters[index]._id, false);
+      }
+
       // Preferred Only update on live
-      farmPrinters[index].options.baudratePreference =
-        settings.connection.preferredBaud;
-      farmPrinters[index].options.portPreference =
-        settings.connection.preferredPort;
-      farmPrinters[index].options.printerProfilePreference =
-        settings.connection.preferredProfile;
+      farmPrinters[index].options = {
+        baudratePreference: settings.connection.preferredBaud,
+        portPreference: settings.connection.preferredPort,
+        printerProfilePreference: settings.connection.preferredProfile
+      };
```

Comment:
This needs blank checking as well currently. Don't see the point at the moment, have added a note for doing in the refactor as the rest needs to be changed too. I'll make the client only send updated values, so the server only has to react to available values.